.github/workflows/nightly.yaml

```diff
@@ -0,0 +1,38 @@
+name: Nightly
+
+on:
+  workflow_dispatch:
+
+permissions:
+  contents: write
+
+jobs:
+  nightly:
+    runs-on: ubuntu-latest
+    if: github.repository == 'playcanvas/engine' && github.ref_name == 'main'
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v4
+
+      - name: Set up Node.js 18.x
+        uses: actions/setup-node@v4
+        with:
+          node-version: 22.x
+          cache: 'npm'
+          registry-url: 'https://registry.npmjs.org/'
+
+      - name: Configure Git User
+        run: |
+          git config --global user.email "action@github.com"
+          git config --global user.name "GitHub Action"
+        shell: bash
+
+      - name: Bump version
```

Comment:
Could we add additional steps to diff check here? Just wary that multiple dispatches will keep bumping without changes.

eg **!!untested!!**

```
- name: Get last tag
  id: tag
  run: |
    echo "tag=$(git describe --tags --abbrev=0)" >> $GITHUB_OUTPUT

- name: Check for changes since last tag
  id: diffcheck
  run: |
    if git diff --quiet ${{ steps.tag.outputs.tag }}..HEAD; then
      echo "changed=false" >> $GITHUB_OUTPUT
    else
      echo "changed=true" >> $GITHUB_OUTPUT
    fi

- name: Bump version
   if: steps.diffcheck.outputs.changed == 'true'
```